kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/NamespaceTest.java

```diff
@@ -0,0 +1,141 @@
+
+  @Test
+  public void testList(){
+    server.expect().withPath("/api/v1/namespaces").andReturn(200, new NamespaceListBuilder()
+      .addNewItem().and()
+      .addNewItem().and()
+      .build()).once();
+
+    KubernetesClient client = server.getClient();
+    NamespaceList namespaceList = client.namespaces().list();
+    assertNotNull(namespaceList);
+    assertEquals(2, namespaceList.getItems().size());
+  }
+
+  @Test
+  public void testListWithLables(){
+    server.expect().withPath("/api/v1/namespaces?labelSelector=" + Utils.toUrlEncoded("key1=value1,key2=value2,key3=value3")).andReturn(200, new NamespaceListBuilder().build()).always();
+    server.expect().withPath("/api/v1/namespaces?labelSelector=" + Utils.toUrlEncoded("key1=value1,key2=value2")).andReturn(200, new NamespaceListBuilder().addNewItem().and()
+      .addNewItem().and()
+      .addNewItem().and()
+      .build()).once();
+
+    KubernetesClient client = server.getClient();
+
+    NamespaceList namespaceList = client.namespaces()
+      .withLabel("key1", "value1")
+      .withLabel("key2", "value2")
+      .list();
+    assertEquals(3, namespaceList.getItems().size());
+
+    namespaceList = client.namespaces()
+      .withLabel("key1", "value1")
+      .withLabel("key2", "value2")
+      .withLabel("key3", "value3")
+      .list();
+    assertEquals(0, namespaceList.getItems().size());
+  }
+
+  @Test(expected = KubernetesClientException.class)
+  public void testEditMissing() {
+    server.expect().withPath("/api/v1/namespaces/namespace1").andReturn(404, "error message from kubernetes").always();
+    KubernetesClient client = server.getClient();
+
+    client.namespaces().withName("namespace1").edit();
+  }
+
+  @Test
+  public void testGet(){
+    server.expect().withPath("/api/v1/namespaces/namespace1").andReturn(200, new NamespaceBuilder().build()).once();
+    server.expect().withPath("/api/v1/namespaces/namespace2").andReturn(200, new NamespaceBuilder().build()).once();
+
+    KubernetesClient client = server.getClient();
+    Namespace namespace = client.namespaces().withName("namespace1").get();
+    assertNotNull(namespace);
+
+    namespace = client.namespaces().withName("namespace2").get();
+    assertNotNull(namespace);
+
+    namespace = client.namespaces().withName("namespace2").get();
+    assertNull(namespace);
+  }
+
+  @Test
+  public void testDelete(){
+    server.expect().withPath("/api/v1/namespaces/namespace1").andReturn(200, new NamespaceBuilder().build()).once();
+    server.expect().withPath("/api/v1/namespaces/namespace2").andReturn(200, new NamespaceBuilder().build()).once();
+
+    KubernetesClient client = server.getClient();
+    Boolean deleted = client.namespaces().withName("namespace1").delete();
+    assertTrue(deleted);
+
+    deleted = client.namespaces().withName("namespace2").delete();
+    assertTrue(deleted);
+
+    deleted = client.namespaces().withName("namespace2").delete();
```

Comment:
why are you querying server twice when you've mocked response on that link for only once?

Reply:
Oh, my bad